cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -69,6 +74,14 @@ func CreateSelfHostedControlPlane(cfg *kubeadmapi.MasterConfiguration, client *c
 		return err
 	}
 
+	if err := launchSelfHostedProxy(cfg, client); err != nil {
```

Comment:
It needs the proxy to exist so that various control plane components (API server) can access etcd through its service